kr8s/_objects.py

```diff
@@ -124,7 +124,9 @@ def api(self, value):
     @property
     def raw(self) -> str:
         """Raw object returned from the Kubernetes API."""
-        return self._raw
+        raw_spec = {"kind": self.kind, "apiVersion": self.version}
+        raw_spec.update(self._raw)
+        return raw_spec
```

Comment:
hi @jacobtomlinson, I think this breaks backward compatibility unintentionally 🤔. Like, after this change we are forced to use `._raw` field for updating single fields.
``` python
# earlier
obj.raw['spec'] = old_obj.raw['spec']

# now
obj._raw['spec'] = old_obj.raw['spec']
```

Even though the intention is to have `raw` attr as getter only, this change might break existing scripts which are riding on this behavior?

Reply:
I'm not sure I understant, there is also a `raw` setter?

https://github.com/kr8s-org/kr8s/blob/03abab8f50e162c866a3b6d54e05957c98fff814/kr8s/_objects.py#L131-L133

Reply:
I believe that setter only works for `obj.raw = value` but not `obj.raw[key] = value`, pls disregard if I'm wrong, I'll recheck if I'm missing something. thanks.

Reply:
Yep that's true! I'm going to make another quick PR to resolve this.

Reply:
It looks like the setter works just fine. I've opened #405 to add a test to verify this is the case. So I think you're wrong in this case.

Reply:
@jacobtomlinson pardon for dragging it, I just gave a try and not sure why the test is working 🤔 
``` python
>>> import kr8s
>>> from kr8s.objects import Pod
>>> kr8s.__version__
'0.16.0'
>>> old_pod = Pod.list()[0]
>>> new_pod = Pod("new_pod")
>>> new_pod.raw
{'kind': 'Pod', 'apiVersion': 'v1', 'metadata': {'name': 'new_pod'}}
>>> len(old_pod.spec)
16
>>> new_pod.raw['spec'] = old_pod.raw['spec']
>>> new_pod.raw
{'kind': 'Pod', 'apiVersion': 'v1', 'metadata': {'name': 'new_pod'}}
>>> 
# ---------------------------------------------------------------------
>>> import kr8s
>>> from kr8s.objects import Pod
>>> kr8s.__version__
'0.15.0'
>>> old_pod = Pod.list()[0]
>>> new_pod = Pod("new_pod")
>>> new_pod.raw
{'metadata': {'name': 'new_pod'}}
>>> len(old_pod.spec)
16
>>> new_pod.raw['spec'] = old_pod.raw['spec']
>>> len(new_pod.spec)
16
>>> new_pod.spec.containers[0].name
'ubi9-bk'
>>>
```
Will take a look at the test whenever possible. thanks.

Reply:
Could you write this up as a test that clearly shows things failing and make a PR with that? Then we can debug further.

Reply:
sure, I'll get it at the earliest, pls excuse any delays.